pkg/mover/chart.go

```diff
@@ -160,13 +163,8 @@ func NewChartMover(req *ChartMoveRequest, opts ...Option) (*ChartMover, error) {
 		return nil, err
 	}
 
-	if req.Source.Chart.Local != nil {
-		err := cm.loadChart(req.Source.Chart.Local.Path)
-		if err != nil {
-			return nil, err
-		}
-	} else {
-		return nil, fmt.Errorf("must provide a local chart as input")
+	if err := cm.loadChart(&req.Source); err != nil {
```

Comment:
I remember in a previous patch that loadChart happened earlier in the function, should we put it back where it was?

Reply:
The move was on purpose. Loading the chart is relatively expensive and I moved forward some validations we want to fail fast first. Why load the chart if other inputs, simpler to check are already wrong?

Anyway, I will review and see what makes more sense.

Reply:
~~~go
func NewChartMover(req *ChartMoveRequest, opts ...Option) (*ChartMover, error) {
	sourceAuth := req.Source.Containers.ContainerRepository
	targetAuth := req.Target.Containers.ContainerRepository
	cm := &ChartMover{
		logger:                  defaultLogger{},
		retries:                 DefaultRetries,
		sourceContainerRegistry: internal.NewContainerRegistryClient(sourceAuth),
		targetContainerRegistry: internal.NewContainerRegistryClient(targetAuth),
	}

	if err := validateTarget(&req.Target); err != nil {
		return nil, err
	}

	if err := cm.loadChart(&req.Source); err != nil {
		return nil, err
	}
~~~
I do not see it.

This code does 3 things:
1. Instantiate the `cm` struct. That needs to happen before loading the chart because we load on the struct. Not sure if it makes sense to remove any of the assignments done there.
2. Validate the request target. That is a quick check and does not make sense to process the whole chart if the inputs are already wrong.
3. The chart is loaded.